test/service/test_request_validation.py

```diff
@@ -236,45 +268,203 @@ def test_file_order(self):
+
+    def test_extra_params(self):
+        url = furl(url=self.base_url, path='/index/files/').url
+        response = requests.get(url, params={'foo': 'bar'})
+        self.assertEqual(400, response.status_code, response.json())
+        self.assertEqual(1, len(response.json()['extra_parameters']))
+        self.assertIn('foo', response.json()['extra_parameters'])
 
     def test_bad_catalog_param(self):
         url = furl(url=self.base_url, path='/index/files').url
         for catalog, error in [
-            ('foo', "Catalog name 'foo' is invalid."),
-            ('foo bar', "Catalog name 'foo bar' contains invalid characters.")
+            ('foo', 'Invalid parameter'),
+            ('foo bar', 'Invalid characters within parameter')
         ]:
             response = requests.get(url=url, params={'catalog': catalog})
             self.assertEqual(400, response.status_code, response.json())
-            self.assertIn(error, response.json()['Message'])
+            self.assertEqual(1, len(response.json()['invalid_parameters']))
+            self.assertIn(error, response.json()['invalid_parameters'][0]['message'])
+
+    def test_deterministic_response(self):
+        url = furl(url=self.base_url, path='/index/files').url
+        shuffle_parameters = [
+            ('filters', '{"}'),
+            ('catalog', 'foo'),
+            ('sort', 'bar'),
+            ('order', 'asc')]
+        shuffle(shuffle_parameters)
+        params = {parameter[0]: parameter[1] for parameter in shuffle_parameters}
+        response = requests.get(url=url, params=params)
+        self.assertEqual(400, response.status_code, response.json())
+        self.assertEqual(3, len(response.json()['invalid_parameters']))
+        self.assertEqual('catalog', response.json()['invalid_parameters'][0]['name'])
+        self.assertEqual('filters', response.json()['invalid_parameters'][1]['name'])
+        self.assertEqual('sort', response.json()['invalid_parameters'][2]['name'])
+
+    @unittest.skip('https://github.com/DataBiosphere/azul/issues/2465')
+    def test_missing_uuid(self):
+        url = furl(url=self.base_url, path='/fetch/repository/files/').url
+        response = requests.get(url, params={'replica': 'aws'})
+        self.assertEqual(400, response.status_code)
+
+    def test_default_for_missing_params(self):
+        path = '/test/mock/endpoints'
+        method_spec = {'parameters': [],
+                       'responses': {'200': {'description': 'OK'}}, 'tags': ['Index']}
+
+        for required, catalog in product((True, False), (self.catalog, None)):
+            param_spec = {'in': 'query', 'schema': {
+                'type': 'string',
+                'pattern': '^([a-z0-9]{1,64})$',
+                'enum': ['test']
+            }, 'required': required, 'name': f'required-{required}-default-{catalog}'}
+            if catalog is not None:
+                param_spec['schema']['default'] = catalog
```

Comment:
I think Hannes generally prefers the declarative style for building dict literals like this. Also notice that I changed the dict literal formatting.
```suggestion
            param_spec = {
                'in': 'query',
                'schema': {
                    'type': 'string',
                    'pattern': '^([a-z0-9]{1,64})$',
                    'enum': ['test']
                },
                'required': required, 
                'name': f'required-{required}-default-{catalog}'
                **({} if catalog is None else {'default': catalog})
            }
```

Reply:
This had to be modified slightly, but 👍🏽